common/thrift_router.h

```diff
@@ -385,34 +384,31 @@ class ThriftRouter {
             v.push_back(hi.first);
           }
         }
-        MoveUpLocalAndRotateRemote(&v, rotation_counter);
+        RankHostsByGroupPrefixLength(&v, rotation_counter, segment);
       }
       return v;
     }
 
-    void MoveUpLocalAndRotateRemote(
+    void RankHostsByGroupPrefixLength(
         std::vector<const Host*>* v,
-        const unsigned rotation_counter) {
-      auto comparator = [this] (const Host* h1, const Host* h2) -> bool {
-        if (h1->az != h2->az &&
-            (h1->az == this->local_az_ || h2->az == this->local_az_)) {
-          return h1->az == this->local_az_;
+        const unsigned rotation_counter,
+        const std::string& segment) {
+      auto comparator = [this, &segment, &rotation_counter]
```

Comment:
remove "&" before rotation_counter. it's more efficient to pass an unsigned variable by copy.